python/python/lance/torch/data.py

```diff
@@ -36,7 +36,7 @@
 
 def _to_tensor(batch: pa.RecordBatch) -> Union[dict[str, torch.Tensor], torch.Tensor]:
     ret = {}
-    for col in batch.column_names:
+    for col in batch.schema.names:
```

Comment:
This is to support older versions of `pyarrow`. `pyarrow` 12.0.0 doesn't have `RecordBatch.column_names`. (For some reason that's the version conda solved for when I needed to install pytorch.)